ecs-cli/modules/cli/compose/entity/service/service.go

```diff
@@ -117,20 +119,31 @@ func (s *Service) LoadContext() error {
 	// specified do not exist.
 	// TODO: Add validation on targetGroupArn or loadBalancerName being
 	// present if containerName or containerPort are specified
-	if role != "" || targetGroupArn != "" || loadBalancerName != "" || containerName != "" || containerPort != nil {
+	if role != "" || targetGroupArn != "" || loadBalancerName != "" || containerName != "" || containerPort != nil || len(targetGroupsList) != 0 {
 		if targetGroupArn != "" && loadBalancerName != "" {
 			return errors.Errorf("[--%s] and [--%s] flags cannot both be specified", flags.LoadBalancerNameFlag, flags.TargetGroupArnFlag)
+		} else if (targetGroupArn != "" || loadBalancerName != "") && len(targetGroupsList) != 0 {
+			return errors.Errorf("[--%s] or [--%s] flags cannot be specified with %s", flags.LoadBalancerNameFlag, flags.TargetGroupArnFlag, flags.TargetGroupsFlag)
```

Comment:
it seems like we don't need to do this actually. I think it is ok to have both old flags and the new flag, since we can just concatenate them.

according to this doc https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-compose-service-up.html. we should have the following logic.
```suggestion
if targetGroupArn != "" || loadBalancerName != "" || containerName != "" || containerPort != nil {
  if role == "" {
    return error (role is needed)
  }
  if containerName == "" {
    return error (container name is required)
  }
  if containerPort == nil {
    return error
  }
  if targetGroupArn != "" && loadBalancerName != "" {
    return error
  }
  if targetGroupArn == "" && loadBalancerName == "" {
    return error
  }
  do sth
}
if len(targetGroupsList) != 0 {
  if role == "" {
    return error
  }
  do sth
}
```